docs/functional/architecture/0001-Field_View_Frontend_Design.md

```diff
@@ -87,7 +84,38 @@ This step makes use of the type information on expressions, for example to decid
 ##### Assume FOAST is correct
 As per guiding principle (5), the lowering should not worry if the FOAST is incorrect or invalid. It is the responsibility of the previous parser and passes to sanitize the user code.
 
+### Program AST (PAST)
+
+The Program AST is a stateful dialect whose main purpose is to apply operators to fields and storing the result on another field. In the future it could be extended to also support control flow (loops, conditionals).
+
+#### What could be changed
+
+The syntax to call field operators feels unnatural as their signature suggests they return a Field, but instead after decorating they return nothing and have an additional argument `out`.
```

Comment:
Just a question: do they really return nothing? In NumPy, a reference to the output value is return even if the `out` parameter is specified.

Reply:
Yes, the reasoning was that `Program`s also return nothing. We could change that easily though. What do you think?

Reply:
If it's easy to change and doesn't conflict with other design decisions, I would return the reference to the output fields since it feels more coherent and matches NumPy behavior.

Reply:
Should be easy, I'll implement it later.